### PR TITLE
[config] Error out when Spake2+ passcode is changed alone

### PR DESCRIPTION
**Pull request by Damian-Nordic**
#### Problem
If one changes the test Spake2+ passcode and leaves the verifier unchanged, PASE stops working which causes confusion among people who are not aware that we already use a pre-generated verifier in the code.

#### Change overview
Add an informative error message when only the passcode is changed to help people realize what needs to be done.

#### Testing
CI only.

### CODE REVIEW

File: src/include/platform/CHIPDeviceConfig.h [hunk 1]
```diff
@@ -955,6 +955,16 @@
...
 #ifndef CHIP_DEVICE_CONFIG_USE_TEST_SPAKE2P_VERIFIER
+
+#if CHIP_DEVICE_CONFIG_USE_TEST_SETUP_PIN_CODE != 20202021
+#error "Non-default Spake2+ passcode configured but verifier left unchanged"
+#endif
+
+#if CHIP_DEVICE_CONFIG_USE_TEST_SPAKE2P_ITERATION_COUNT != 1000
+#error "Non-default Spake2+ iteration count configured but verifier left unchanged"
+#endif
+
```
Comment (emargolis) [inline src/include/platform/CHIPDeviceConfig.h:966]: Need to add similar check for CHIP_DEVICE_CONFIG_USE_TEST_SPAKE2P_SALT

File: src/include/platform/CHIPDeviceConfig.h [hunk 2]
```diff
@@ -955,6 +955,16 @@
  *   The value is base-64 encoded string.
  */
 #ifndef CHIP_DEVICE_CONFIG_USE_TEST_SPAKE2P_VERIFIER
+
+#if CHIP_DEVICE_CONFIG_USE_TEST_SETUP_PIN_CODE != 20202021
+#error "Non-default Spake2+ passcode configured but verifier left unchanged"
```
Comment (bzbarsky-apple) [inline src/include/platform/CHIPDeviceConfig.h:960]: What will happen if none of the CHIP_DEVICE_CONFIG_USE_TEST* bits are defined at all?  Will that `#if` test true of false?

Comment (Damian-Nordic) [reply src/include/platform/CHIPDeviceConfig.h:960]: If they are not defined by a user, they are set to defaults right above, so they are always defined at this place.

File: src/include/platform/CHIPDeviceConfig.h (hunk 1 above)
Comment (Damian-Nordic) [reply src/include/platform/CHIPDeviceConfig.h:966]: Right, but you can't compare strings in a preprocessor. I can add a constexpr function + static assert though.

File: src/include/platform/CHIPDeviceConfig.h (hunk 2 above)
Comment (jmartinez-silabs) [reply src/include/platform/CHIPDeviceConfig.h:961]: This works in a case where the setup pin code is not provided in tthe nvm key else `CHIP_DEVICE_CONFIG_USE_TEST_SETUP_PIN_CODE` is unchanged but not used.

```
err = mGenericConfigManager.ReadConfigValue(ConfigClass::kConfigKey_SetupPinCode, setupPasscode);
#if defined(CHIP_DEVICE_CONFIG_USE_TEST_SETUP_PIN_CODE) && CHIP_DEVICE_CONFIG_USE_TEST_SETUP_PIN_CODE
    if (err == CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND)
    {
        setupPasscode = CHIP_DEVICE_CONFIG_USE_TEST_SETUP_PIN_CODE;
        err           = CHIP_NO_ERROR;
    }
#endif // defined(CHIP_DEVICE_CONFIG_USE_TEST_SETUP_PIN_CODE) && CHIP_DEVICE_CONFIG_USE_TEST_SETUP_PIN_CODE
    SuccessOrExit(err);

```

**Review by bzbarsky-apple — APPROVED** [review]

**Review by emargolis — APPROVED** [review]

File: src/include/platform/CHIPDeviceConfig.h (hunk 2 above)
Comment (Damian-Nordic) [reply src/include/platform/CHIPDeviceConfig.h:961]: Yes, but if someone stores the passcode in the nvm or elsewhere, is there a reason that they keep `CHIP_DEVICE_CONFIG_USE_TEST_SETUP_PIN_CODE` defined? I simply wanted to help people who simply play around with examples in the repo. I assume that if someone is familiar with the code enough to provide their own implementation or nvm data, they will know about the verifier, right?

**Pull request merged**